external/avro/src/main/scala/org/apache/spark/sql/avro/functions.scala

```diff
@@ -72,6 +72,19 @@ object functions {
    */
   @Experimental
   def to_avro(data: Column): Column = {
-    new Column(CatalystDataToAvro(data.expr))
+    new Column(CatalystDataToAvro(data.expr, None))
+  }
```

Comment:
If we have the default value `None`, we don't need to touch this function.

Reply:
See my comment in https://github.com/apache/spark/pull/25419#discussion_r313206434